### PR TITLE
servermaster(engine): generate master id based on name

### PR DESCRIPTION
**Pull request by gozssky**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tiflow/issues/7177

### What is changed and how it works?

Generate a master id with the prefix of name.

#### Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

### Release note 

```release-note
None
```

### CODE REVIEW

File: engine/servermaster/service_util.go [hunk 1]
```diff
@@ -42,3 +50,9 @@ func newMultiClient(conn *grpc.ClientConn) multiClient {
 		JobManagerClient:      enginepb.NewJobManagerClient(conn),
 	}
 }
+
+func generateNodeID(name string) string {
+	val := rand.Uint32()
+	id := fmt.Sprintf("%s-%08x", name, val)
```
Comment (lonng) [inline engine/servermaster/service_util.go:56]: What if the name is empty?

**Review by lonng — COMMENTED** [review]
Rest LGTM

File: deployments/engine/docker-compose/1m1e.yaml [hunk 2]
```diff
@@ -9,6 +9,7 @@ services:
     command:
       - "/tiflow"
       - "master"
+      - "--name=server-master"
```
Comment (CharlesCheung96) [inline deployments/engine/docker-compose/1m1e.yaml:12]: Why remove the configuration here?

Comment (gozssky) [reply deployments/engine/docker-compose/1m1e.yaml:12]: "remove" -> "add" ?

The default name is `"server-master-" + $advertise_addr` which is too long for the name.

File: engine/servermaster/service_util.go (hunk 1 above)
Comment (gozssky) [reply engine/servermaster/service_util.go:56]: The master and executor will generate a default name for themselves, so the name will not be empty.

**Comment by gozssky** [comment]
/run-engine-integration-test

MySQL is too slow:
```
[2022/09/22 07:21:22.697 +00:00] [ERROR] [master.go:87] ["run dataflow server master with error"] [error="[DFLOW:ErrMetaOpFail]meta operation fail: context deadline exceeded"]
```

**Review by CharlesCheung96 — APPROVED** [review]

**Review by amyangfei — APPROVED** [review]

**Pull request merged**